lib/internal/blob.js

```diff
@@ -395,13 +396,18 @@ function ClonedBlob() {
 }
 ClonedBlob.prototype[kDeserialize] = () => {};
 
+function TransferrableBlob(handle, length, type = '') {
```

Comment:
this change is observable:

```mjs
structuredClone(new Blob([])).constructor // TransferrableBlob
structuredClone(new Blob([])).constructor === Blob // false, should be true
```

Reply:
I didn't change the behavior of `structuredClone`, the `ClonedBlob` is responsible for creating and maintaining the `Blob` after `structuredClone`.

I tried:

```js
const {Blob} = require('buffer');

console.log(structuredClone(new Blob([])).constructor);
console.log(structuredClone(new Blob([])).constructor === Blob);
```

![image](https://github.com/nodejs/node/assets/12551007/98135183-b1ca-4fbd-aaae-d716bf10b03e)

Reply:
my bad, createBlob isn't used in ClonedBlob. However it's still an issue with anything that uses createBlob:

```js
> new Blob(['a']).slice(0, 1).constructor
[Function: TransferrableBlob]
```

Reply:
You are right, fixed using [this trick](https://github.com/nodejs/node/pull/49620/files#diff-a75b6b92eada0f88527acce4428a28471b39067fd0ddaab2fcf26cc022bded47R1235-R1239), thanks @rluvaton :)

I also added tests to cover this issue.

Reply:
Please re-run benchmarks after the fix just in case :)